### ISSUE TITLE
MDL and MOL export must correctly annotate bond block tables

### ISSUE THREAD

**Issue by tobigithub** [grimme-lab/xtb#158]
**Describe the bug**
The MDL and SDF export for xtb currently only exports the atom types and connections between atoms, the bond tables are all set to "1" which means single bond. That will lead to very wrong results once people import and re-use optimized MOL and SDF files. Basically once the InchiKey changes from input to output file,  the connections and atom types all subsequent calculations are wrong.

All cheminformatics editors obey the bond table rules that designate for example aromatic or kekule type aromatic. Most of the excellent cheminformatics packages to that too. For example here is an explanation contributed by Robert Belford Professor (Chemistry) at University of Arkansas at Little Rock).  [[LINK]](https://chem.libretexts.org/Courses/University_of_Arkansas_Little_Rock/ChemInformatics_(2017)%3A_Chem_4399%2F%2F5399/2.2%3A_Chemical_Representations_on_Computer%3A_Part_II/2.2.2%3A_Anatomy_of_a_MOL_file) and [[MDL MOL Definitions]](http://c4.cabrillo.edu/404/ctfile.pdf)

![image](https://user-images.githubusercontent.com/2569565/77370576-69d16b00-6d1e-11ea-97be-7a6ae56db02e.png)

As of xtb version 6.3.0 I would not recommend to use any MOL or SDF export, but only XYZ. 
The reason is that SDF/MOL is a very well established format, people will simply assume that there are no issues (there are issues with aromaticity and handling of nitro compounds etc among different packages). However simple bond types always should be correct.

![image](https://user-images.githubusercontent.com/2569565/77372304-aacb7e80-6d22-11ea-9538-7481375df375.png)

The results are not good, the compound is now a completely different compound. This does not happen with other packages that use forcefields, here the bond types are always retained.

![image](https://user-images.githubusercontent.com/2569565/77372363-ccc50100-6d22-11ea-974e-1d2df5e6f323.png)

**To Reproduce**
Run example molecule with single, double, triple and aromatic bond types (2,4) and check output.

Please provide all input and output file such that we confirm your report.

**Expected behaviour**
No change in Inchikey and connection tables, except for few issues (elongated bonds), heteroatoms etc.

**Additional context**

1) One way **for the user to avoid that** currently is to use only XYZ files, because the connection table is missing anyway and has to be restored with openbabel or other tools (xyz2mol). For example  obabel -ixyz -omol result.xyz > result.mol   In many cases the mol file is correct, not in all cases.

2) Another **way to programmatically avoid that** is to **keep the bond order** from the input connection table MOL/SDF file (bottom part) and **only replace the atom coordinates** (top part) in the MOL/SDF file. In this case the atoms should not be reordered.

In very few cases this will lead to errors. So we can see by just updating the XYZ coordinates, but not the atom types we effectively solved this severe issue. The second part of the inchikey is changed because the stereochemistry changed, but that is could expected for a 3D optimization. In this case 16 stereoisomers would be possible.

I think if the second fix is applied it could solve the issue, there are probably other cases (nitro groups, aromatic bonds such as type 4) that are not that easy, but that maybe another issue. Overall we can see the input mol matches the output mol. The explanation of bond block tables is here [[on page 14]](http://c4.cabrillo.edu/404/ctfile.pdf).

![image](https://user-images.githubusercontent.com/2569565/77373263-00a12600-6d25-11ea-968c-c317d23cf367.png)

**Comment by awvwgk** [comment]
Thanks for the detailed report, `xtb` indeed only saves the connectivity, but drops the bond orders.

Option 1 is to actually save all the bond information and just echo them, which will be wrong if we break a bond. Option 2 would be to regenerate the bond order information from the Wiberg bond orders to give the correct bond orders, which could also lead to a change in the bond topology.

I'm in favour of option 2, because we already implemented option 1 wrong.

**Comment by ghutchis** [comment]
Since you have the Wiberg bond orders, I think it's reasonable to take that approach.

Both OB and Jan's code perform a set of analysis.. it's maybe not the best idea in your case (i.e., why implement it in `xtb` when other packages exist).

Inevitably, some users will complain of 'wrong' bond orders, so I think the WBO analysis is best here - the answer is simple - `xtb` uses a quantum analysis whether it fits your intuition or not.

I don't know if you store the fractional bond orders in an MDL property, but that would also be useful since users could make their own decisions on borderline cases.

IMHO,  I'd love to constrain CREST to avoid breaking bonds, since it's broken off some fragments, but that's a different issue.

**Comment by ghutchis** [comment]
Incidentally, we are finishing a random-forest classifier for bond orders - it gets high accuracy.. probably a few more weeks given current disruptions before we upload to GitHub.

**Comment by awvwgk** [comment]
The quick fix is in, for everything else, move the discussion to #34. Since I turning the `xtb` code base upside down right now, it might take a few more days before we can safely merge new features to master.
